pkg/patterns/addon/pkg/status/kstatus.go

```diff
@@ -70,22 +75,66 @@ func (k *kstatusAggregator) BuildStatus(ctx context.Context, info *declarative.S
 				log.Info("resource status was nil")
 				statusMap[status.UnknownStatus] = true
 			}
+			abnormalTrueConds := getAbnormalTrueConditions(ctx, unstruct)
+			totalAbnormalTrueConditions = append(totalAbnormalTrueConditions, abnormalTrueConds...)
 		}
 
-		aggregatedPhase := string(aggregateStatus(statusMap))
-
-		if currentStatus.Phase != aggregatedPhase {
-			currentStatus.Phase = aggregatedPhase
+		// Summarize all the deployment manifests statuses to a single results.
+		aggregatedPhase := aggregateStatus(statusMap)
+		// Update the Conditions for the declarativeObject status.
+		if aggregatedPhase == status.CurrentStatus {
+			SetReady(&currentStatus, totalAbnormalTrueConditions)
+		} else {
+			SetInProgress(&currentStatus, totalAbnormalTrueConditions)
 		}
+		currentStatus.Phase = string(aggregatedPhase)
 	}
 
 	if err := utils.SetCommonStatus(info.Subject, currentStatus); err != nil {
 		return err
 	}
+	return nil
+}
+
+// getAbnormalTrueConditions calculates the abnormal-true conditions that best describe the current state of the deployment manifests.
+func getAbnormalTrueConditions(ctx context.Context, unstruct *unstructured.Unstructured) []status.Condition {
+	log := log.FromContext(ctx)
```

Comment:
As another example, I think you could do

```
log := log.FromContext(ctx).WithValues("object", unstruct)
```

And then you would know that object would be in all your log messages in this function.

It's only two instances here, so it doesn't matter, but just wanted to try to clarify when you might use this!